main_test.go

```diff
@@ -510,6 +515,74 @@ var testCases = []TestCase{
 	},
 }
 
+type FakeS3Client struct {
+	Get mockGetObjectAPI
+	Put mockPutObjectAPI
+}
+
+type mockGetObjectAPI func(ctx context.Context, params *s3.GetObjectInput) (*s3.GetObjectOutput, error)
+type mockPutObjectAPI func(ctx context.Context, params *s3.PutObjectInput) (*s3.PutObjectOutput, error)
+
+func (fc *FakeS3Client) GetObject(ctx context.Context, params *s3.GetObjectInput, optFns ...func(*s3.Options)) (*s3.GetObjectOutput, error) {
+	return fc.Get(ctx, params)
+}
+
+func (fc *FakeS3Client) PutObject(ctx context.Context, params *s3.PutObjectInput, optFns ...func(*s3.Options)) (*s3.PutObjectOutput, error) {
+	return fc.Put(ctx, params)
+}
+
+func (fc *FakeS3Client) New(filename string) *FakeS3Client {
+	return &FakeS3Client{
+		Get: mockGetObjectAPI(func(ctx context.Context, params *s3.GetObjectInput) (*s3.GetObjectOutput, error) {
+			bucket := params.Bucket
+			if bucket == nil {
+				return nil, errors.New("bucket is nil")
+			}
+			key := params.Key
+			if key == nil {
+				return nil, errors.New("key is nil")
+			}
+			f, err := excelize.OpenFile(filename)
```

Comment:
why not just do `os.Open` on the file and set the resulting `File` object as the body of the response?

Reply:
Yes, that works. My thought was to do more checking and get the number of bytes downloaded. Done.